scipy/sparse/linalg/_eigen/lobpcg/lobpcg.py

```diff
@@ -598,10 +630,19 @@ def lobpcg(
         lambdaHistory[0, :] = _lambda
 
     eigBlockVector = np.asarray(eigBlockVector[:, ii])
-    blockVectorX = np.dot(blockVectorX, eigBlockVector)
-    blockVectorAX = np.dot(blockVectorAX, eigBlockVector)
+    blockVectorX = _dot_inplace(
+        blockVectorX, eigBlockVector,
```

Comment:
This signature is weird to me, i.e. assigning the output of a function that says "inplace". Why not go all the way, like:
```
_dot_inplace(blockVectorX, eigBlockVector, verbosityLevel=verbosityLevel)  # updates blockVectorX in place
```

Reply:
To my recollection, the proposed change would lead to errors in situations where in-place calculation is impossible due to dtype mismatches.

Reply:
OK, fair enough.